java8/src/main/java/net/nycjava/java8/api/J8OptionalApi.java

```diff
@@ -1,41 +1,23 @@
 package net.nycjava.java8.api;
 
-import java.util.AbstractMap.SimpleEntry;
-import java.util.List;
-import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-import static java.util.Arrays.asList;
-import static java.util.Collections.unmodifiableMap;
-import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.joining;
 
 public class J8OptionalApi {
 
 
     public static void main(String[] args) {
+        System.out.println(concatName("Douglas", null, "Crockford"));
```

Comment:
I don't know if he does or doesn't have a middle name. If you are sure he doesn't, then this makes sense, otherwise find someone who doesn't, or else go with a fictitious name

Reply:
Sure, I'll go with fictitious names throughout